Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -0,0 +1,600 @@
+
+        var proxy = imageryProvider._proxy;
+        if (defined(proxy)) {
+            url = proxy.getURL(url);
+        }
+
+        return url;
+    }
+
+    defineProperties(UrlTemplateImageryProvider.prototype, {
+        /**
+         * The pattern to request a tile which has the following keywords:
+         * <ul>
+         *  <li> <code>{z}</code>: The level of the tile in the tiling scheme.  Level zero is the root of the quadtree pyramid.</li>
+         *  <li> <code>{x}</code>: The tile X coordinate in the tiling scheme, where 0 is the Westernmost tile.</li>
+         *  <li> <code>{y}</code>: The tile Y coordinate in the tiling scheme, where 0 is the Northernmost tile.</li>
+         *  <li> <code>{s}</code>: One of the available subdomains, used to overcome browser limits on the number of simultaneous requests per host.</li>
+         *  <li> <code>{reverseX}</code>: The tile X coordinate in the tiling scheme, where 0 is the Easternmost tile.</li>
+         *  <li> <code>{reverseY}</code>: The tile Y coordinate in the tiling scheme, where 0 is the Southernmost tile.</li>
+         *  <li> <code>{westDegrees}</code>: The Western edge of the tile in geodetic degrees.</li>
+         *  <li> <code>{southDegrees}</code>: The Southern edge of the tile in geodetic degrees.</li>
+         *  <li> <code>{eastDegrees}</code>: The Eastern edge of the tile in geodetic degrees.</li>
+         *  <li> <code>{northDegrees}</code>: The Northern edge of the tile in geodetic degrees.</li>
+         *  <li> <code>{westProjected}</code>: The Western edge of the tile in projected coordinates of the tiling scheme.</li>
+         *  <li> <code>{southProjected}</code>: The Southern edge of the tile in projected coordinates of the tiling scheme.</li>
+         *  <li> <code>{eastProjected}</code>: The Eastern edge of the tile in projected coordinates of the tiling scheme.</li>
+         *  <li> <code>{northProjected}</code>: The Northern edge of the tile in projected coordinates of the tiling scheme.</li>
+         * </ul>
+         * @memberof UrlTemplateImageryProvider.prototype
+         * @type {String}
+         * @readonly
+         */
+        url : {
+            get : function() {
+                return this._url;
+            }
+        },
+
+        /**
+         * Gets the proxy used by this provider.
+         * @memberof UrlTemplateImageryProvider.prototype
+         * @type {Proxy}
+         * @readonly
+         */
+        proxy : {
+            get : function() {
+                return this._proxy;
+            }
+        },
+
+        /**
+         * Gets the width of each tile, in pixels. This function should
+         * not be called before {@link UrlTemplateImageryProvider#ready} returns true.
+         * @memberof UrlTemplateImageryProvider.prototype
+         * @type {Number}
+         * @readonly
+         */
+        tileWidth : {
+            get : function() {
+                //>>includeStart('debug', pragmas.debug);
```

Comment:
Do we need to throw all of these exceptions, e.g., `_tileWidth` is set in the constructor and not touched elsewhere.  Is this a carryover from another imagery provider or am I missing something?